src/pages/SettingsView/components/CollectorSettings/domain/useCollectorSettings.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  DeleteCollectionRuleRequest,
+  GetCollectionRuleResponse,
+  UpsertCollectionRuleRequest,
+} from '@buf/pyroscope_api.bufbuild_es/settings/v1/setting_pb';
+import { displayError, displaySuccess } from '@shared/domain/displayStatus';
+import { DomainHookReturnValueTyped as DomainHookReturnValue } from '@shared/types/DomainHookReturnValue';
+import { useEffect, useMemo, useState } from 'react';
+
+import { useFetchCollectorRules } from './useFetchCollectorRules';
+
+export function useCollectorSettings(): DomainHookReturnValue<
+  {
+    rules: GetCollectionRuleResponse[];
+    existingRuleNames: string[];
+    isFetching: boolean;
+    fetchError: Error | null;
+  },
+  {
+    saveRule(rule: UpsertCollectionRuleRequest): Promise<void>;
+    deleteRule(ruleName: string): Promise<void>;
+  }
+> {
+  const { rules, isFetching, error: fetchError, upsertAsync, deleteAsync } = useFetchCollectorRules();
+
+  const [currentState, setCurrentState] = useState<GetCollectionRuleResponse[]>(rules ?? []);
+  const existingRuleNames = useMemo(() => currentState.map((r) => r.name), [currentState]);
+
+  // TODO: useMemo, because it's not a side effect
```

Comment:
I don't understand that, I want/need the rules to be set from the inital request, but then want to update them as we save/delete rules. How do I achieve this with `useMemo`?